eclass/rocm.eclass

```diff
@@ -0,0 +1,217 @@
+#     >=sci-libs/rocBLAS-${ROCM_VERSION}[${ROCM_USEDEP}] )"
+#
+# src_configure() {
+#     if use rocm; then
+#         local amdgpu_flags=$(get_amdgpu_flags)
+#         export HCC_AMDGPU_TARGET=${amdgpu_flags//;/,}
+#     fi
+#     default
+# }
+# src_test() {
+#     use rocm && check_amdgpu
+#     default
+# }
+# @CODE
+
+if [[ ! ${_ROCM_ECLASS} ]]; then
+
+case ${EAPI} in
+	7|8) ;;
+	*) die "${ECLASS}: EAPI ${EAPI:-0} not supported" ;;
+esac
+
+# @ECLASS_VARIABLE: ROCM_VERSION
+# @REQUIRED
+# @PRE_INHERIT
+# @DESCRIPTION:
+# The ROCm version of current package. For ROCm libraries, it should be ${PV};
+# for other packages that depend on ROCm libraries, this can be set to match
+# the version required for ROCm libraries.
+
+# @ECLASS_VARIABLE: ROCM_REQUIRED_USE
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Requires at least one AMDGPU target to be compiled.
+# Example use for ROCm libraries:
+# @CODE
+# REQUIRED_USE="${ROCM_REQUIRED_USE}"
+# @CODE
+# Example use for packages that depend on ROCm libraries:
+# @CODE
+# IUSE="rocm"
+# REQUIRED_USE="rocm? ( ${ROCM_REQUIRED_USE} )"
+# @CODE
+
+# @ECLASS_VARIABLE: ROCM_USEDEP
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# This is an eclass-generated USE-dependency string which can be used to
+# depend on another ROCm package being built for the same AMDGPU architecture.
+#
+# The generated USE-flag list is compatible with packages using rocm.eclass.
+#
+# Example use:
+# @CODE
+# DEPEND="sci-libs/rocBLAS[${ROCM_USEDEP}]"
+# @CODE
+
+# @FUNCTION: _rocm_set_globals
+# @DESCRIPTION:
+# Set global variables useful to ebuilds: IUSE, ROCM_REQUIRED_USE, and
+# ROCM_USEDEP
+_rocm_set_globals() {
+	# Two lists of AMDGPU_TARGETS of certain ROCm version.  Official support
+	# matrix:
+	# https://docs.amd.com/bundle/ROCm-Installation-Guide-v${ROCM_VERSION}/page/Prerequisite_Actions.html.
+	# There is no well-known unofficial support matrix.
+	# https://github.com/Bengt/ROCm/blob/patch-2/README.md#library-target-matrix
+	# may help. Gentoo have patches to enable gfx1031 as well.
+	local unofficial_amdgpu_targets official_amdgpu_targets
+	case ${ROCM_VERSION} in
+		4.*)
+			unofficial_amdgpu_targets=(
+				gfx803 gfx900 gfx1010 gfx1011 gfx1012 gfx1030
+			)
+			official_amdgpu_targets=(
+				gfx906 gfx908
+			)
+			;;
+		5.*)
+			unofficial_amdgpu_targets=(
+				gfx803 gfx900 gfx1010 gfx1011 gfx1012 gfx1031
+			)
+			official_amdgpu_targets=(
+				gfx906 gfx908 gfx90a gfx1030
+			)
+			;;
+		*)
+			die "Unknown ROCm major version! Please update rocm.eclass before bumping to new ebuilds"
+			;;
+	esac
+
+	local iuse_flags=( "${official_amdgpu_targets[@]/#/+amdgpu_targets_}" "${unofficial_amdgpu_targets[@]/#/amdgpu_targets_}" )
+	IUSE="${iuse_flags[*]}"
+
+	local all_amdgpu_targets=( "${official_amdgpu_targets[@]}" "${unofficial_amdgpu_targets[@]}" )
+	local allflags=( "${all_amdgpu_targets[@]/#/amdgpu_targets_}" )
+	ROCM_REQUIRED_USE=" || ( ${allflags[*]} )"
+
+	local optflags=${flags[@]/%/(-)?}
```

Comment:
```suggestion
	local optflags=${allflags[@]/%/(-)?}
```

?

Reply:
Look like I wrote stupid code again, and it escapes my quick test suite. Thanks!